src/gui/attributetable/qgsdualview.cpp

```diff
@@ -943,13 +954,45 @@ void QgsDualView::resizeColumn()
   }
 }
 
+void QgsDualView::resizeAllColumns()
+{
+  QAction *action = qobject_cast<QAction *>( sender() );
+  int col = action->data().toInt();
+  if ( col < 0 )
+    return;
+
+  QgsAttributeTableConfig config = mConfig;
+
+  bool ok = false;
+  int width = QInputDialog::getInt( this, tr( "Set column width" ), tr( "Enter column width" ),
+                                      mTableView->columnWidth( col ),
+                                      1, 1000, 10, &ok );
+  if ( ok )
+  {
+    int col_count = mTableView->model()->columnCount();
+    if ( col_count > 0)
+    {
+      for ( int i=0; i<col_count; i++ )
+      {
+        config.setColumnWidth( i, width );
+      }
+      setAttributeTableConfig( config );
+    }
+  }
+}
+
 void QgsDualView::autosizeColumn()
 {
   QAction *action = qobject_cast<QAction *>( sender() );
   int col = action->data().toInt();
   mTableView->resizeColumnToContents( col );
 }
 
+void QgsDualView::autosizeAllColumns()
+{
+    mTableView->resizeColumnsToContents();
```

Comment:
```suggestion
  mTableView->resizeColumnsToContents();
```

Reply:
@nyalldawson, thank you very much for the review and changes!